dffml/record.py

```diff
@@ -112,11 +112,13 @@ class Record(object):
 
     def __init__(
         self,
-        key: str,
+        key: str = None,
```

Comment:
Records require a key, let's remove the `= None` which makes it an optional keyword argument